### PR TITLE
feature(bigquery): improve error guidance for streaming inserts

### PR DESCRIPTION
**Pull request by shollyman**
This PR attempts to clarify guidance for users of the streaming insert mechanism.

BigQuery enforces a variety of limitations on client requests (size, qps, etc).  When
those limits are more severely violated, BigQuery replies with less specific information
about the issue, and instead generates an unstructured HTTP 400 client error response.

This PR adds guidance to doc.go about this behavior, and adds an integration test that
induces these errors to ensure the error surface is behaving as documented.

Fixes: #2855

### CODE REVIEW

File: bigquery/inserter.go [hunk 1]
```diff
@@ -19,16 +19,28 @@ import (
...
 
+// ErrRequestMalformed is surfaced for cases where the response from the backend
+// indicates a generic error, but is unstructured and ambiguous.
+//
+// For streaming insert requests, this often indicates that either the row
+// data is excessively large, or that the batch factor is too high (too many rows
+// sent as part of a single insert).  However, the error response provides no
+// further diagnostic information, and may indicate problems other than size such
+// as QPS or fundamental problems with the payload.
+var ErrRequestMalformed = fmt.Errorf("your client has issued a malformed or illegal request")
```
Comment (hongalex) [inline bigquery/inserter.go:42]: prefer `errors.New` if you aren't using wrapping

**Review by hongalex — APPROVED** [review]

File: bigquery/inserter.go (hunk 1 above)
Comment (codyoss) [reply bigquery/inserter.go:42]: Do you expect people to type switch on this error and handle this case specially? Just trying to understand if this needs to be exported.

Comment (shollyman) [reply bigquery/inserter.go:42]: I'm coming around to the viewpoint of not adding a custom error, and simply augmenting the package documentation to be more explicit.  It provides little value and doesn't reduce ambiguity sufficiently.

**Comment by shollyman** [comment]
Nothing further has bubbled up in discussions, so I'm going to move forward with committing this as it  doesn't yield any changes in the library contracts; it merely attempts to improve documentation and testing.

**Pull request merged**